Add reading streams from media to FileLoader and MediaService

JoshuaBehrens opened the pull request:
### 1. Why is this change necessary?

When you want to access the filesystem and need to process big files, you want a stream and not a string.

### 2. What does this change do, exactly?

Expose stream related methods from the flysystem API.

### 3. Describe each step to reproduce the issue or behaviour.

1. Make a file download for private media
2. Request a file that is bigger in size than your allowed php memory limit
3. php-fpm wins "best titanic impression" contest

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

JoshuaBehrens commented:
Squash and changelog will get in here later

shyim commented on src/Core/Content/Media/MediaService.php:
```diff
@@ -114,6 +115,11 @@ public function loadFile(string $mediaId, Context $context): string
         return $this->fileLoader->loadMediaFile($mediaId, $context);
     }
 
+    public function loadFileStream(string $mediaId, Context $context): StreamInterface
```
Can you add tests for this new method?

JoshuaBehrens committed:
Add loadMediaFileStream to FileLoader

Add service dependency to a stream factory
Add loadFileStream to MediaService

JoshuaBehrens commented:
I added a test to the FileLoader as the media service just passes it through. I also added a changelog and rebased.

I was the first writing FileLoader tests?!

mstegmeyer commented:
Your PR has been merged with 094534d2ec25eb08de7b9273ee05e5fdef323ee4 with some adjustments in the tests, since they were non-functional 🤐 .

Thank you for your contribution 🥳 💙

JoshuaBehrens commented:
Thank you a lot for fixing it before I noticed ☺️
